File: find_quiz_api.js
const fs = require('fs');

async function main() {
    const API_BASE_URL = "https://api-mankai.staging.rikkei.edu.vn/v1";
    const EMAIL = "maiquoctuan1994@gmail.com";
    const PASSWORD = "Tuan@2012";

    const res = await fetch(`${API_BASE_URL}/auth/login`, {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ email: EMAIL, password: PASSWORD })
    });
    const loginData = await res.json();
    const token = loginData.data.accessToken;

    const data = JSON.parse(fs.readFileSync('mankai_data.json', 'utf8'));
    const quiz_lesson = data.lessons.find(l => l.type === 'QUIZ');

    if (quiz_lesson) {
        const api_id = quiz_lesson.api_id;
        console.log(`Testing QUIZ API for lesson ${api_id}: ${quiz_lesson.title}`);

        const endpoints = [
            `${API_BASE_URL}/student/lessons/${api_id}`,
            `${API_BASE_URL}/quiz/${api_id}`,
            `${API_BASE_URL}/exam/${api_id}`,
            `${API_BASE_URL}/student/exam/${api_id}`,
            `${API_BASE_URL}/student/exam?lessonId=${api_id}`,
        ];

        for (const ep of endpoints) {
            console.log(`GET ${ep}`);
            try {
                const r = await fetch(ep, {
                    headers: { 'Authorization': `Bearer ${token}` }
                });
                console.log(`Status: ${r.status}`);
                if (r.status === 200) {
                    const text = await r.text();
                    console.log(text.substring(0, 200));
                }
            } catch (e) {
                console.log(e);
            }
        }
    } else {
        console.log("No QUIZ found");
    }
}

main();

File: find_quiz_api.py
import requests
import json

API_BASE_URL = "https://api-mankai.staging.rikkei.edu.vn/v1"
EMAIL = "maiquoctuan1994@gmail.com"
PASSWORD = "Tuan@2012"

session = requests.Session()
res = session.post(f"{API_BASE_URL}/auth/login", json={"email": EMAIL, "password": PASSWORD})
token = res.json()["data"]["accessToken"]
session.headers.update({"Authorization": f"Bearer {token}"})

# Known quiz lesson ID from n1_structure.txt
# e.g., "Test ngày 1_08/09/2025" in "90 NGÀY ĐÊM MANTEN N1"
# We need to find its api_id. Let's load mankai_data.json
with open("mankai_data.json", "r", encoding="utf-8") as f:
    data = json.load(f)

quiz_lesson = None
for l in data["lessons"]:
    if l["type"] == "QUIZ":
        quiz_lesson = l
        break

if quiz_lesson:
    api_id = quiz_lesson["api_id"]
    print(f"Testing QUIZ API for lesson {api_id}: {quiz_lesson['title']}")
    
    endpoints = [
        f"{API_BASE_URL}/student/lessons/{api_id}",
        f"{API_BASE_URL}/quiz/{api_id}",
        f"{API_BASE_URL}/exam/{api_id}",
        f"{API_BASE_URL}/exams/{api_id}",
        f"{API_BASE_URL}/lesson-quiz/{api_id}",
        f"{API_BASE_URL}/student/exams?lessonId={api_id}",
    ]
    
    for ep in endpoints:
        print(f"GET {ep}")
        try:
            r = session.get(ep)
            print(f"Status: {r.status_code}")
            if r.status_code == 200:
                print(str(r.json())[:200])
        except Exception as e:
            print(e)
else:
    print("No QUIZ found")
